app/app.go

```diff
@@ -9,276 +9,255 @@ import (
-
-		// total supply of the chain
 		supply.AppModuleBasic{},
-
-		// inflation
-		mint.AppModuleBasic{},
-
+		genutil.AppModuleBasic{},
+		bank.AppModuleBasic{},
 		staking.AppModuleBasic{},
-
-		slashing.AppModuleBasic{},
-
+		mint.AppModuleBasic{},
 		distr.AppModuleBasic{},
-
+		gov.NewAppModuleBasic(
+			paramsclient.ProposalHandler, distr.ProposalHandler,
+		),
 		params.AppModuleBasic{},
+		crisis.AppModuleBasic{},
+		slashing.AppModuleBasic{},
+		evidence.AppModuleBasic{},
 
 		// akash
 		deployment.AppModuleBasic{},
 		market.AppModuleBasic{},
 		provider.AppModuleBasic{},
 	)
+
+	// module account permissions
+	maccPerms = map[string][]string{
+		auth.FeeCollectorName:     nil,
+		distr.ModuleName:          nil,
+		mint.ModuleName:           {supply.Minter},
+		staking.BondedPoolName:    {supply.Burner, supply.Staking},
+		staking.NotBondedPoolName: {supply.Burner, supply.Staking},
+		gov.ModuleName:            {supply.Burner},
+	}
+
+	// module accounts that are allowed to receive tokens
+	allowedReceivingModAcc = map[string]bool{
+		distr.ModuleName: true,
+	}
 )
 
 // AkashApp extends ABCI appplication
 type AkashApp struct {
 	*bam.BaseApp
 	cdc *codec.Codec
 
+	invCheckPeriod uint
+
+	// keys to access the substores
 	keys  map[string]*sdk.KVStoreKey
 	tkeys map[string]*sdk.TransientStoreKey
 
-	keeper struct {
-		acct       auth.AccountKeeper
-		bank       bank.Keeper
-		params     params.Keeper
-		supply     supply.Keeper
-		staking    staking.Keeper
-		distr      distr.Keeper
-		slashing   slashing.Keeper
-		mint       mint.Keeper
-		deployment deployment.Keeper
-		market     market.Keeper
-		provider   provider.Keeper
-	}
-
+	// subspaces
+	subspaces map[string]params.Subspace
+
+	// keepers
+	AccountKeeper    auth.AccountKeeper
+	BankKeeper       bank.Keeper
+	SupplyKeeper     supply.Keeper
+	StakingKeeper    staking.Keeper
+	SlashingKeeper   slashing.Keeper
+	MintKeeper       mint.Keeper
+	DistrKeeper      distr.Keeper
+	GovKeeper        gov.Keeper
+	CrisisKeeper     crisis.Keeper
+	ParamsKeeper     params.Keeper
+	EvidenceKeeper   evidence.Keeper
+	DeploymentKeeper deployment.Keeper
+	MarketKeeper     market.Keeper
+	ProviderKeeper   provider.Keeper
+
+	// the module manager
 	mm *module.Manager
 }
 
-// ModuleBasics returns all app modules basics
-func ModuleBasics() module.BasicManager {
-	return mbasics
-}
-
 // MakeCodec returns registered codecs
 func MakeCodec() *codec.Codec {
 	var cdc = codec.New()
-
-	mbasics.RegisterCodec(cdc)
-
+	ModuleBasics.RegisterCodec(cdc)
+	vesting.RegisterCodec(cdc)
 	sdk.RegisterCodec(cdc)
 	codec.RegisterCrypto(cdc)
-	codec.RegisterEvidences(cdc)
-
 	return cdc.Seal()
 }
 
 // https://github.com/cosmos/sdk-tutorials/blob/c6754a1e313eb1ed973c5c91dcc606f2fd288811/app.go#L73
 
-// NewApp creates and returns a new Akash App.
-func NewApp(
-	logger log.Logger, db dbm.DB, tio io.Writer, options ...func(*bam.BaseApp),
+// NewAkashApp creates and returns a new Akash App.
+func NewAkashApp(
```

Comment:
Why are you changing the name?

Reply:
because the constructor name should match the returned value (i.e `AkashApp`)

Reply:
okay.  would have preferred to change the struct name to `App`.